Session.js

```diff
@@ -49,21 +49,33 @@ export default class Session {
     await this.refresh();
   }
 
+  /**
+   * Registers a handler that is executed when an event listener fires.
+   *
+   * @param {string} eventType - An event such as change.
+   * @param {function} handler - A handler function called when
+   *   an event occurs.
+   *
+   * @returns {function} A unique function to remover the listener.
+   */ 
   on(eventType, handler) {
     if(typeof eventType !== 'string') {
       throw new TypeError('"eventType" must be a string.');
     }
     if(typeof handler !== 'function') {
       throw new TypeError('"handler" must be a function.');
     }
-
+    // this should return a Map for the eventType.
     const listeners = this._eventTypeListeners.get(eventType);
     if(!listeners) {
       throw new Error(`Event "${eventType}" is not supported.`);
     }
 
     const remover = () => listeners.delete(remover);
+    // use the function as a unique key in the event listener Map.
```

Comment:
```suggestion
    // use the function as a unique key in the event listener Map
```

Reply:
@mattcollier has a big thing about proper punctuation so I think the `.` should stay.